var/spack/repos/builtin/packages/cgm/package.py

```diff
@@ -25,30 +25,51 @@
 from spack import *
 
 
-class Cgm(Package):
+class Cgm(AutotoolsPackage):
     """The Common Geometry Module, Argonne (CGMA) is a code library
        which provides geometry functionality used for mesh generation and
        other applications."""
-    homepage = "http://trac.mcs.anl.gov/projects/ITAPS/wiki/CGM"
-    url      = "http://ftp.mcs.anl.gov/pub/fathom/cgm13.1.1.tar.gz"
+    homepage = "http://sigma.mcs.anl.gov/cgm-library"         
+    url = "http://ftp.mcs.anl.gov/pub/fathom/cgm-16.0.tar.gz"
 
+    version('16.0', 'a68aa5954d82502ff75d5eb91a29a01c')
     version('13.1.1', '4e8dbc4ba8f65767b29f985f7a23b01f')
     version('13.1.0', 'a6c7b22660f164ce893fb974f9cb2028')
     version('13.1', '95f724bda04919fc76818a5b7bc0b4ed')
 
-    depends_on("mpi")
+    variant("mpi", default=False, description='enable mpi support')
+    variant("oce", default=False, description='enable oce geometry kernel')
+
+    depends_on('mpi', when='+mpi')
+    depends_on('oce+X11', when='+oce')
+
+    def url_for_version(self, version):
+        if Version('13.0') <= version <= Version('13.999'):
+            return "http://ftp.mcs.anl.gov/pub/fathom/cgm{0}.tar.gz".format(version)
+        else:
+            return "http://ftp.mcs.anl.gov/pub/fathom/cgm-{0}.tar.gz".format(version)
```

Comment:
It's not too late to rename your download tarballs if you want to make things simpler 😆